olive/passes/olive_pass.py

```diff
@@ -211,14 +215,58 @@ def run(
                 )
                 components.append(output_model_component)
                 component_names.append(component_name)
+                Pass._carry_forward_additional_files(component_model, output_model_component)
             output_model = CompositeModelHandler(components, component_names)
         else:
             output_model = self._run_for_config(model, data_root, config, output_model_path)
+            Pass._carry_forward_additional_files(model, output_model)
+
         # assumption: the model attributes from passes, if any, are more important than
         # the input model attributes, we should not update/extend anymore outside of the pass run
         output_model.model_attributes = output_model.model_attributes or model.model_attributes
         return output_model
 
+    @staticmethod
+    def _carry_forward_additional_files(input_model: OliveModelHandler, output_model: OliveModelHandler):
+        # NOTE: Can't use model.model_path because that always gets resolved to a filepath.
+        # We need the directory path here.
+        input_model_path = input_model.get_resource("model_path")
+        if not input_model_path:
+            return
+
+        input_model_path = Path(input_model_path)
+        if not input_model_path.is_dir():
+            return
+
+        input_model_attributes = input_model.model_attributes or {}
+        input_model_additional_files = set(input_model_attributes.get("additional_files", []))
+        if not input_model_additional_files:
+            return
+
+        output_model_path = Path(output_model.get_resource("model_path"))
+        if not output_model_path.is_dir() and (output_model_path.suffix == ".onnx"):
+            # change the "model_path" resource to the parent directory of the model file
+            output_model.set_resource("model_path", output_model_path.parent)
+            output_model.onnx_file_name = output_model_path.name
+            output_model_path = output_model_path.parent
```

Comment:
i think an elif to still check for `not output_model_path.is_dir()` is still required? The if else only handles onnx models. 
It could be a value error like before or a warning and return.

Reply:
Done!